src/main/java/com/arpnetworking/metrics/common/kafka/ConsumerDeserializer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Dropbox.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.common.kafka;
+
+import com.arpnetworking.commons.jackson.databind.ObjectMapperFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Jackson <code>JsonDeserializer</code> implementation for <code>Consumer</code>.
+ *
+ * @param <K> the type of key field in <code>Consumer</code>
+ * @param <V> the type of value field in <code>Consumer</code>
+ *
+ * @author Joey Jackson (jjackson at dropbox dot com)
+ */
+public class ConsumerDeserializer<K, V> extends JsonDeserializer<Consumer<K, V>> {
+
+    @Override
+    public Consumer<K, V> deserialize(final JsonParser parser, final DeserializationContext context)
+            throws IOException, JsonProcessingException {
+        // Parse input json into JsonNode Tree
+        final JsonDeserializer<? extends JsonNode> deserializer = JsonNodeDeserializer.getDeserializer(ObjectNode.class);
+        final JsonNode node = deserializer.deserialize(parser, context);
+
+        // Pull out configs and topics fields and convert deserialize with standard mapper
+        final ObjectMapper mapper = ObjectMapperFactory.getInstance();
+        final TypeReference<Map<String, String>> configsType = new TypeReference<Map<String, String>>() {};
+        final Map<String, Object> configs = mapper.convertValue(node.get("configs"), configsType);
```

Comment:
nit: It looks like `node.get("configs")` will silently return `null` if `node` is not a JSON object. To avoid this, can we make `node` an `ObjectNode` instead of a generic `JsonNode`?